src/kubelet/client/connector.go

```diff
@@ -0,0 +1,252 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/transport"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+)
+
+const (
+	apiProxyPath = "/api/v1/nodes/%s/proxy/"
+	httpSchema   = "http"
+	httpsSchema  = "https"
+)
+
+type Connector interface {
+	connect() (*connParams, error)
+}
+
+type defaultConnector struct {
+	// TODO: Use a non-sdk logger
+	logger             log.Logger
+	apiServerHost      string
+	kc                 kubernetes.Interface
+	tripperBearerToken http.RoundTripper
+	config             config.Mock
+}
+
+func (dp defaultConnector) connect() (*connParams, error) {
```

Comment:
`defaultConnector` is also reasonably big, I'd consider using a pointer receiver to avoid copying too much.